src/psyclone/domain/lfric/psyir.py

```diff
@@ -66,7 +68,7 @@
         MODULE_NAME.upper(), MODULE_NAME))
     # Create the variables specified by the module (using PSyIR DataSymbols)
     for module_var in module.vars:
-        exec("{0} = DataSymbol('{1}', DeferredType(), interface="
+        exec("{0} = DataSymbol('{1}', INTEGER_TYPE, interface="
```

Comment:
f-strings please.

Reply:
I am still getting a few pylint warnings about f-strings (sorry, 'you touched it' policy), one `super()` without args.

And messages about undefined variables `i_def` etc (which is probably ok, since they are created here).

Reply:
Yes, they are created here so I've disabled the warnings and fixed the f-strings. (As an aside we may be able to do away with `exec` (see https://stackoverflow.com/questions/100003/what-are-metaclasses-in-python) but that's out of scope here.)

Reply:
Thanks. I've written a long rant here about exec, but decided to move this to #1777 :)